daffodil-runtime1/src/main/scala/org/apache/daffodil/dpath/ConverterOps.scala

```diff
@@ -179,159 +136,80 @@ case object DoubleToBoolean extends Converter {
-    val res = a.getLong
-    checkRange(res)
-    asBigInt(res)
-  }
-  override protected val rangePrim = PrimType.UnsignedLong
+case object LongToUnsignedLong extends Converter {
+  override def computeValue(a: DataValuePrimitive, dstate: DState): DataValueBigInt = PrimType.UnsignedLong.fromNumber(a.getLong).getBigInt
 }
 
 case object NumericToDouble extends Converter {
   override def computeValue(a: DataValuePrimitive, dstate: DState): DataValueDouble = asDouble(a.getAnyRef)
 }
 
 case object StringToBoolean extends Converter {
-  override def computeValue(a: DataValuePrimitive, dstate: DState): DataValueBool = {
-    val str = a.getString
-    val res =
-      if (str == "true" || str == "1") true
-      else if (str == "false" || str == "0") false
-      else throw new NumberFormatException("Value '%s' is not a valid boolean value {true, false, 1, 0}.".format(str))
-    asBoolean(res)
-  }
+  override def computeValue(a: DataValuePrimitive, dstate: DState): DataValueBool = PrimType.Boolean.fromXMLString(a.getString).getBoolean
 }
 case object StringToDecimal extends Converter {
-  override def computeValue(a: DataValuePrimitive, dstate: DState): DataValueBigDecimal = new JBigDecimal(a.getString)
+  override def computeValue(a: DataValuePrimitive, dstate: DState): DataValueBigDecimal = PrimType.Decimal.fromXMLString(a.getString).getBigDecimal
 }
 case object StringToDouble extends Converter {
-  override def computeValue(a: DataValuePrimitive, dstate: DState): DataValueDouble = {
-    val str = a.getString
-    val d =
-      if (str == XMLUtils.PositiveInfinityString) JDouble.POSITIVE_INFINITY
-      else if (str == XMLUtils.NegativeInfinityString) JDouble.NEGATIVE_INFINITY
-      else if (str == XMLUtils.NaNString) JDouble.NaN
-      else str.toDouble
-    d
-  }
+  override def computeValue(a: DataValuePrimitive, dstate: DState): DataValueDouble = PrimType.Double.fromXMLString(a.getString).getDouble
 }
 case object StringToLong extends Converter {
-  override def computeValue(a: DataValuePrimitive, dstate: DState): DataValueLong = {
-    val res =
-      try {
-        a.getString.toLong
-      } catch {
-        case nfe: NumberFormatException => {
-          val e = new NumberFormatException("Cannot convert to type long: " + nfe.getMessage())
-          throw e
-        }
-      }
-    res
-  }
+  override def computeValue(a: DataValuePrimitive, dstate: DState): DataValueLong = PrimType.Long.fromXMLString(a.getString).getLong
 }
-case object StringToUnsignedLong extends Converter with NumericRangeCheck {
-  override def computeValue(a: DataValuePrimitive, dstate: DState): DataValueBigInt = {
-    val res = new JBigInt(a.getString)
-    checkRange(res)
-    res
-  }
-  override protected val rangePrim = PrimType.UnsignedLong
+case object StringToUnsignedLong extends Converter {
+  override def computeValue(a: DataValuePrimitive, dstate: DState): DataValueBigInt = PrimType.UnsignedLong.fromXMLString(a.getString).getBigInt
```

Comment:
I think this is a great refactoring, but the lines are now really long. Could you maybe move the PrimType... to a new line, e.g.:
```suggestion
  override def computeValue(a: DataValuePrimitive, dstate: DState): DataValueBigInt = 
    PrimType.UnsignedLong.fromXMLString(a.getString).getBigInt
```

Reply:
Missed addressing this comment. I think it would be worth wrapping these long lines.